src/collision/CollisionHandler.cpp

```diff
@@ -134,37 +140,39 @@ void CollisionHandler::detectLineCollision(
             The x distance to the end point from the origin.
         const int deltaY,
             The y distance to the end point from the origin.
-        std::priority_queue<Target>& targetsInSights,
-            This is the queue that will hold the targets that will be determined.
+        TargetList &targetList,
+            This is the priority queue wrapper that will hold the targets that will be determined.
         std::vector<Entity*> allEntities,
             The entities that will need to be searched for valid targets.
         int type
             the type of entity, see target Target.h for definitions.
             This is needed for identification purposes in InstantWeapon.fire()
 */
-void CollisionHandler::checkTargets(const int originX, const int originY, const int deltaX, const int deltaY,
-        std::priority_queue<Target>& targetsInSights, std::vector<Entity*>& allEntities, int type) {
-
-    int tempOriginX = originX;
-    int tempOriginY = originY;
-    int tempEndX = originX + deltaX;
-    int tempEndY = originY + deltaY;
-    int distanceToOrigin;
+void CollisionHandler::checkTargets(const int originX, const int originY, const int endX, const int endY,
```

Comment:
This method needs documentation
After doing my own research on the SDL_IntersectRectAndLine call I realized why you moved the temporaries, and why those temporaries exist in the first place
I shouldn't have to do that
I had already set up several comments chewing you out for making code worse by moving temporaries into a loop and using them in ways that made no sense
This code is the opposite of self-documenting, and you need to fix it
Tell me why there are temporary variables, why they aren't const or references, and the implications that the SDL intersect rect call inside this method has upon those temporary vars
Because at a first glance, this code will look odd to anyone